src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,460 @@
+                    textBuffer: textBuffer,
+                    method: method,
+                    capabilitiesFilter: capabilitiesFilter,
+                    languageServerName: languageServerName,
+                    parameterFactory: ParameterFactory,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                if (response is not null && response.Response is not null)
+                {
+                    await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                    var body = response.Response.ToObject<DocumentSymbol[]>();
+                    var docSymbols = DocumentOutlineHelper.GetDocumentSymbols(body);
+                    this.originalTree = docSymbols;
+                    symbolTree.ItemsSource = docSymbols;
+                }
+                else
+                {
+                    symbolTree.ItemsSource = new List<DocSymbol>();
+                }
+            }
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+        {
+            var documentSymbols = new List<DocSymbol>();
+            foreach (var item in (List<DocSymbol>)symbolTree.ItemsSource)
+            {
+                documentSymbols.Add(ExpandAllNodes(item));
+            }
+
+            symbolTree.ItemsSource = documentSymbols;
+        }
+
+        private DocSymbol ExpandAllNodes(DocSymbol treeItem)
+        {
+            treeItem.IsExpanded = true;
+            foreach (var childItem in treeItem.Children.OfType<DocSymbol>())
+            {
+                ExpandAllNodes(childItem);
+            }
+
+            return treeItem;
+        }
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+        {
+            var documentSymbols = new List<DocSymbol>();
+            foreach (var item in (List<DocSymbol>)symbolTree.ItemsSource)
+            {
+                documentSymbols.Add(CollapseAllNodes(item));
+            }
+
+            symbolTree.ItemsSource = documentSymbols;
+        }
+
+        private DocSymbol CollapseAllNodes(DocSymbol treeItem)
+        {
+            treeItem.IsExpanded = false;
+            foreach (var childItem in treeItem.Children.OfType<DocSymbol>())
+            {
+                CollapseAllNodes(childItem);
+            }
+
+            return treeItem;
+        }
+
+        private void RemoveText(object sender, EventArgs e)
+        {
+            if (searchBox.Text == "Search Document Outline")
+            {
+                searchBox.Text = "";
+            }
+        }
+
+        private void AddText(object sender, EventArgs e)
+        {
+            if (string.IsNullOrWhiteSpace(searchBox.Text))
+                searchBox.Text = "Search Document Outline";
+        }
+
+        private void Search(object sender, EventArgs e)
+        {
+            if (searchBox.Text == "Search Document Outline" || string.IsNullOrWhiteSpace(searchBox.Text))
```

Comment:
we can't rely on the text being in English. instead we may want to have Boolean value to determine if the text is default or (as I described above) define a new resource and reference it (`ServicesVSResources.Search_Document_Outline`) here instead